doc/specifications/proposed/named_queries/developer_documentation.md

```diff
@@ -0,0 +1,149 @@
+
+```twig
+match:
+  QueryType\Parameters:
+    type: "article",
+    category: "development"
+
+Each parameter will be matched exactly.
+The matcher will match if ALL of the provided parameters match what is contained in the hash.
+
+##### `QueryType\Expression`
+Uses Symfony's [Expression Language](http://symfony.com/en/doc/current/components/expression_language/index.html)
+for advanced matching.
+
+This matcher expects a valid expression language string as the input. The expression must be
+evaluated to a boolean. The parameters hash is available as `parameters`
+
+In the example below, 'type' must be either 'article' or 'blog_post'.
+
+```twig
+match:
+  QueryType\Expression: "parameters['type'] in ['article', 'blog_post']"
+```
+
+## QueryType objects
+To make a new QueryType available to the Query Controller, you need to register a new
+QueryType and register it.
+
+A QueryType is a PHP class that implements QueryType interface (`eZ\Publish\Core\MVC\Query\QueryType`).
+It has two methods:
+- `getName()`: returns the QueryType name
+- `getQuery( array $parameters = null )`: returns the contained Query.
+  It accepts an optional hash of parameters that can be used in any way in the implementation:
+  - customize an element's value (limit, ContentType identifier, etc)
+  - conditionally add/remove criteria from the query
+  - set the limit/offset...
+
+
+### QueryType example: latest content
+This QueryType contains a Query that returns the 10 last published content, order by reverse
+publishing date. It accepts an optional `ContentType` parameter, that can be set to a ContentType
+identifier:
+
+```php
+<?php
+namespace AcmeBundle\eZ\Query;
+
+use eZ\Publish\Core\MVC\Query\QueryType;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
+
+class LatestContentQueryType implements QueryType
+{
+  public function getQuery( array $parameters = [] )
+  {
+    $query = new LocationQuery();
+    $query->criterion = new Criterion\LogicalAnd([
+      $criteria[] = new Criterion\Visibility( Criterion\Visibility::VISIBLE );
+      $criteria[] = new Criterion\ContentTypeIdentifier( 'article' );
+      $criteria[] = new Criterion\ParentLocationId(2);
+      $criteria[] = new Criterion\Location\Depth( Criterion\Operator:GT, 1 )
+    ]);
+    $query->sortClauses = [new SortClause\DatePublished()];
+    $query->limit = 10;
+
+    return $query;
+  }
+
+  public function getName()
+  {
+    return 'LatestContent';
+  }
+}
+```
+
+### Registering a QueryType
+In addition to creating a PHP class for a `QueryType`, it must also be registered
+within the Service Container. This can be done in two ways: by convention, and
+with a service tag.
+
+#### By convention
+Any class named `AcmeBundle\eZ\QueryType\*QueryType`, that implements
+the QueryType interface, will be registered as a
+custom QueryType.
+
+#### Using a service tag
+```yaml
+acme.query.latest_content:
+    class: AcmeBundle\Query\LatestContent
+    tags:
+        - {name: ez_query}
```

Comment:
Maybe let's follow the same (implicit) conventions than for other tags?

I'd suggest `ezpublish.query_type`